qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1511B.py

```diff
@@ -0,0 +1,63 @@
+from typing import Optional, TYPE_CHECKING, List
+
+from .KeysightB1517A import B1517A
+from .constants import IMeasRange, IOutputRange
+
+if TYPE_CHECKING:
+    from .KeysightB1500_base import KeysightB1500
+
+
+class B1511B(B1517A):
+    """
+    Driver for Keysight B1511B Source/Monitor Unit module for B1500
+    Semiconductor Parameter Analyzer.
+
+    Args:
+        parent: mainframe B1500 instance that this module belongs to
+        name: Name of the instrument instance to create. If `None`
+            (Default), then the name is autogenerated from the instrument
+            class.
+        slot_nr: Slot number of this module (not channel number)
+        asu_present: Flag to acknowledge ASU presence
+    """
+
+    def __init__(self, parent: 'KeysightB1500', name: Optional[str],
+                 slot_nr: int, asu_present: bool = False, **kwargs):
```

Comment:
assuming ASU is present, how would users make sure asu_present argument here is set to True?

in the current implementation, user does not instantiate modules explicitly, instead B1500 class does that. So in the current implementation there's no way the user can instantiate B1500 with B1511 and ASU present.

I see at least two options: 1) make asu_present an argument to B1500 (and pass it all the way to B1511 init) so that the user can tell B1500 at init time that ASU is present, 2) make asu_present an attribute (or property) of B1511 so that the user can tell B1500/B1511 AFTER init time if ASU is present.

Option 2 seems to me the least amount of effort which is great since the users that we know of don't use ASU.